app/decorators/table_sorters/hearings_sorter.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module TableSorters
+  class HearingsSorter
+    def initialize(hearings, column, direction)
+      @hearings = hearings
+      @column = column
+      @direction = direction
+    end
+
+    def self.for(hearings, column, direction)
+      Hash.new(TableSorters::HearingsDateSorter.new(hearings, column, direction)).merge(
+        type: TableSorters::HearingsTypeSorter.new(hearings, column, direction),
+        provider: TableSorters::HearingsProviderSorter.new(hearings, column, direction)
+      )[column.to_sym]
```

Comment:
You  don't need to repeat the `.new(hearings, column, direction)`

```suggestion
      Hash.new(TableSorters::HearingsDateSorter).merge(
        type: TableSorters::HearingsTypeSorter,
        provider: TableSorters::HearingsProviderSorter
      )[column.to_sym].new(hearings, column, direction)
```

Reply:
have added a commit to this affect